src/kotlinSessions/SessionTwo.kt

```diff
@@ -5,33 +5,36 @@ fun main() {
     //---------------------------------Loops----------------------------------
 
     // a loop is a block which is designed to be of a repeated execution
-    // so instead of saying
-    // println(x + 1)
+    // instead of saying
+    // var x: Int = 0
     // println(x + 1)
+    // println(x + 1) .....
     // we would commonly say
-    println("Test while loop")
+    println("Test for loop")
+    for (number in 0..2 step 1) {
+        println(number)
+    }
+
+    // the same code can be written using a while loop with operand inequalities operations (>=, <=, ==, !=)
+    println("Test equivalent while loop")
+    // var i = 0 and i <= 2 is the same as ordering a looper in the range 0..2
     var i: Int = 0
-    while (i < 2) {
-        i = i + 1
+    while (i <= 2) {
         println(i)
+        // equivalent to step 1 in the previous for loop
+        i = i + 1
     }
 
-    // do while loop is the same as while loop but the first condition is a mandatory condition (ie is executed even if the condition is not true)
+    // do while loop is the same as while loop but the first condition is a mandatory condition .
+    // (ie: it is executed only one time even if the condition is not true).
```

Comment:
Not enough explanation

Reply:
Do you have any insightful information on this ? 

i think may be we should explain a step-by-step comment of what the loop does.